src/core/shader.js

```diff
@@ -96,8 +95,22 @@ Shader.prototype = {
     var schema = this.schema;
     dataKeys.forEach(processData);
     function processData (key) {
+      var materialKey;
       if (!schema[key] || schema[key].is !== type) { return; }
-      if (variables[key].value === data[key]) { return; }
+      if (schema[key].type === 'map') {
+        // Special handling is needed for textures.
+        materialKey = '_texture_' + key;
+        // If data unchanged, get out early.
+        if (!variables[key] || variables[key].value === data[key]) { return; }
+        // We can't actually set the variable correctly until we've loaded the texture.
+        self.el.addEventListener('materialtextureloaded', function (e) {
+          variables[key].value = self.material[materialKey];
+          variables[key].needsUpdate = true;
+        });
+        // Kick off the texture update now that handler is added.
+        utils.material.updateMapMaterialFromData(materialKey, key, self, data);
```

Comment:
it isn't clear to me that will work since not all cases here are just a string url as previously discussed.  if you have some time to try that technique and have video srcObject, canvas, etc. still work, go ahead and take a crack at it, but if not I would like you to consider merging this as-is for now and taking on texture system refactor for 0.5.0